### PR TITLE
RabbitMQ: Add parameter types

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This PR includes;
- Add parameter types
- Fixed one type from 'str' to 'list'
- Clean up EXAMPLES
- Added seealso sections

##### ISSUE TYPE
- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME
rabbitmq modules

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/112238/48/tests

**Comment by dagwieers** [comment]
It seems the included classes do not support something the module expects. It is unrelated to my changes IMO.

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_user.py [hunk 1]
```diff
@@ -131,8 +140,6 @@ def __init__(self, module, username, password, tags, permissions,
         self.node = node
         if not tags:
             self.tags = list()
-        else:
-            self.tags = tags.split(',')
```
Comment (mattclay) [inline lib/ansible/modules/messaging/rabbitmq/rabbitmq_user.py:135]: This change appears to be responsible for the CI failure.

Comment (dagwieers) [reply lib/ansible/modules/messaging/rabbitmq/rabbitmq_user.py:135]: I see, fixed that thinko.

**Comment by dagwieers** [comment]
@gundalow Please review.

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_exchange.py [hunk 2]
```diff
@@ -7,74 +7,74 @@
...
+    default: direct
+    aliases: [ type ]
+  auto_delete:
+    description:
+    - Whether the exchange should delete itself after all queues/exchanges unbound from it.
+    type: bool
+    default: no
+  internal:
+    description:
+    - hether the exchange is available only for other exchanges.
```
Comment (acozine) [inline lib/ansible/modules/messaging/rabbitmq/rabbitmq_exchange.py]:
```suggestion
    - Whether the exchange is available only for other exchanges.
```

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_parameter.py [hunk 3]
```diff
@@ -7,57 +7,65 @@
...
     description:
-      - erlang node name of the rabbit we wish to configure
+    - Erlang node name of the rabbit we wish to configure.
+    type: str
     default: rabbit
     version_added: "1.2"
   state:
     description:
-      - Specify if user is to be added or removed
+    - Specify if user is to be added or removed.
```
Comment (acozine) [inline lib/ansible/modules/messaging/rabbitmq/rabbitmq_parameter.py:51]: Not added by this PR, but I'm wondering why the description here talks about `users`, when the module adds/removes/updates `parameters`. Is this copy-pasta?

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_plugin.py [hunk 4]
```diff
@@ -131,18 +138,18 @@ def disable(self, name):
...
     )
     module = AnsibleModule(
         argument_spec=arg_spec,
-        supports_check_mode=True
+        supports_check_mode=True,
     )
 
     result = dict()
-    names = module.params['names'].split(',')
+    names = module.params['names']
```
Comment (acozine) [inline lib/ansible/modules/messaging/rabbitmq/rabbitmq_plugin.py:152]: I'm don't fully understand the consequences (if any) of removing `.split(',')`.

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_queue.py [hunk 5]
```diff
@@ -7,89 +7,103 @@
...
+author:
+- Manuel Sousa (@manuel-sousa)
 version_added: "2.0"
-
-short_description: Manage rabbitMQ queues
+short_description: Manage RabbitMQ queues
 description:
-  - This module uses rabbitMQ Rest API to create/delete queues
-requirements: [ "requests >= 1.0.0" ]
+- This module uses RabbitMQ Rest API to create/delete queues.
```
Comment (acozine) [inline lib/ansible/modules/messaging/rabbitmq/rabbitmq_queue.py]:
```suggestion
- This module uses the RabbitMQ Rest API to create/delete queues.
```

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_user.py [hunk 6]
```diff
@@ -7,96 +7,105 @@
...
+    - This option should be preferable when you care about all permissions of the user.
+    - You should use vhost, configure_priv, write_priv, and read_priv options instead
+      if you care about permissions for just some vhosts.
+    type: list
     default: []
   vhost:
     description:
-      - vhost to apply access privileges.
-      - This option will be ignored when permissions option is used.
+    - vhost to apply access privileges.
```
Comment (acozine) [inline lib/ansible/modules/messaging/rabbitmq/rabbitmq_user.py]:
```suggestion
    - Vhost to apply access privileges.
```

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_vhost.py [hunk 7]
```diff
@@ -109,15 +115,15 @@ def _disable_tracing(self):
 
 def main():
     arg_spec = dict(
-        name=dict(required=True, aliases=['vhost']),
-        tracing=dict(default='off', aliases=['trace'], type='bool'),
-        state=dict(default='present', choices=['present', 'absent']),
-        node=dict(default='rabbit'),
+        name=dict(type='str', required=True, aliases=['vhost']),
+        tracing=dict(type='bool', default=False, aliases=['trace']),
```
Comment (acozine) [inline lib/ansible/modules/messaging/rabbitmq/rabbitmq_vhost.py:119]: should this be `default=No`?

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_vhost_limits.py [hunk 8]
```diff
@@ -7,75 +7,77 @@
...
+    default: -1
+  node:
+    description:
+    - Name of the RabbitMQ Erlang node to manage.
+  state:
+    description:
+    - Specify whether the limits are to be set or cleared.
+    - If set to C(absent), the limits of both I(max_connections) and I(max-queues) will be cleared.
+    default: present
+    choices: [present, absent]
```
Comment (acozine) [inline lib/ansible/modules/messaging/rabbitmq/rabbitmq_vhost_limits.py]:
```suggestion
    choices: [absent, present]
```

**Review by acozine — COMMENTED** [review]
Found a few nits, but generally the docs changes look good. I'm not sure of the scope of some of the  other changes - have the rabbitmq maintainers reviewed this PR?

**Review by acozine — COMMENTED** [review]
Oops, I made some comments but never finished my review.

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_vhost.py (hunk 7 above)
Comment (dagwieers) [reply lib/ansible/modules/messaging/rabbitmq/rabbitmq_vhost.py:119]: No

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_plugin.py (hunk 4 above)
Comment (dagwieers) [reply lib/ansible/modules/messaging/rabbitmq/rabbitmq_plugin.py:152]: This supports YAML lists. We don't want users to provide a comma-separated string, we want them to use lists. This allows both.

**Pull request closed without merging**